lib/datadog/appsec/component.rb

```diff
@@ -28,6 +29,19 @@ def build_appsec_component(settings, telemetry:)
 
         private
 
+        def incompatible_ffi_version?
+          ffi_version = Gem.loaded_specs['ffi'] && Gem.loaded_specs['ffi'].version
+          return false unless RUBY_VERSION >= '3.3.0' && ffi_version < '1.16.0'
```

Comment:
I think you would need to parse the FFI version correctly, string comparison works for Ruby versions only because their components are kept to single digits.

I would also check against 3.3 not 3.3.0.

Reply:
Added `Gem::Version` for all version comparisons here: [03f3bc9](https://github.com/DataDog/dd-trace-rb/pull/3969/commits/03f3bc90a34dcc748370d295a0635157526f58a9)